src/main.rs

```diff
@@ -76,63 +76,140 @@ impl Machine {
             let current_op_code = self.memory[self.pc];
             let instruction_len = match current_op_code {
 
+                // ==========================
+                // ========== 0x1_ ==========
+                // ==========================
+
                 // load
                 0x10 => {
-                    let src = self.memory[self.pc + 1];
+                    let src = self.memory[self.pc.wrapping_add(1)];
                     self.acc = self.memory[src];
                     2
                 }
 
                 // load immediate
                 0x11 => {
-                    self.acc = self.memory[self.pc + 1];
+                    self.acc = self.memory[self.pc.wrapping_add(1)];
                     2
                 }
 
                 // store
                 0x12 => {
-                    let dst = self.memory[self.pc + 1];
+                    let dst = self.memory[self.pc.wrapping_add(1)];
                     self.memory[dst] = self.acc;
                     2
                 }
 
+                // store immediate
+                0x13 => {
+                    let val = self.memory[self.pc.wrapping_add(1)];
+                    let dst = self.memory[self.pc.wrapping_add(2)];
+                    self.memory[dst] = val;
+                    3
+                }
+
+                // move
+                0x14 => {
+                    let src = self.memory[self.pc.wrapping_add(1)];
+                    let dst = self.memory[self.pc.wrapping_add(2)];
+                    self.memory[dst] = self.memory[src];
+                    3
+                }
+
+                // ==========================
+                // ========== 0x2_ ==========
+                // ==========================
+
                 // jump
                 0x20 => {
-                    self.pc = self.memory[self.pc + 1];
+                    self.pc = self.memory[self.pc.wrapping_add(1)];
                     0
                 }
 
                 // jump zero
                 0x21 => {
                     if self.acc == 0 {
-                        self.pc = self.memory[self.pc + 1];
+                        self.pc = self.memory[self.pc.wrapping_add(1)];
                         0
                     } else {
                         2
                     }
                 }
 
+                // ==========================
+                // ========== 0x3_ ==========
+                // ==========================
+
+                // add
+                0x30 => {
+                    let src = self.memory[self.pc.wrapping_add(1)];
+                    self.acc = self.acc.wrapping_add(self.memory[src]);
+                    2
+                }
+
                 // add immediate
                 0x31 => {
-                    self.acc = self.acc.wrapping_add(self.memory[self.pc + 1]);
+                    self.acc = self.acc.wrapping_add(self.memory[self.pc.wrapping_add(1)]);
```

Comment:
Maybe you want to format this like `add`: store the argument `v` in a variable first? So: two lines instead of one?